x-pack/legacy/plugins/canvas/canvas_plugin_src/uis/arguments/textarea.js

```diff
@@ -18,21 +18,26 @@ const TextAreaArgInput = ({ updateValue, value, confirm, commit, renderError, ar
     return null;
   }
   return (
-    <EuiForm>
-      <EuiTextArea
-        className="canvasTextArea__code"
-        id={argId}
-        rows={10}
-        value={value}
-        resize="none"
-        onChange={confirm ? updateValue : ev => commit(ev.target.value)}
-      />
+    <div>
+      <EuiFormRow display="rowCompressed">
+        <EuiTextArea
+          className="canvasTextArea__code"
+          id={argId}
+          compressed
+          rows={10}
+          value={value}
+          resize="none"
+          onChange={confirm ? updateValue : ev => commit(ev.target.value)}
+        />
+      </EuiFormRow>
       <EuiSpacer size="s" />
-      <EuiButton size="s" onClick={() => commit(value)}>
-        {confirm}
-      </EuiButton>
+      <EuiFormRow display="rowCompressed">
```

Comment:
Button's shouldn't be wrapped in `EuiFormRow`s because the form row is more than just visual. It handles focus states and passes id's and such that's only needed for form inputs.